### DISCUSSION TITLE
Initialize dynamically added content

### DISCUSSION THREAD

**Discussion by peterhijma** [alpinejs/alpine#1584] [7. Bugs (archived)]
Using: Alpine 3.1.0

I'm adding some dynamic html to a modal (using `fetch()`), including the cancel button for example.

So I've got something like this dynamically added:
```
<button @click="$store.modal.close()">Cancel</button>
```

to some div:
```
<div x-html="someContent"></div>
```
The `@click` doesn't work. 

Is there any way to get this to work?

**Comment by peterhijma** [comment]
Hm, for now I've been able to get it working using some magic like this:

```
<element>._x_refs.content.innerHTML = this.html;
```

Is there a nicer method? :)

**Reply by ryangjchandler** [reply]
I'm guessing Alpine isn't walking the new HTML and registering the directives.

**Reply by peterhijma** [reply]
@ryangjchandler  Is it worth an issue, do you think?

**Comment by Luddinus** [comment]
I need the same, posted here: https://github.com/alpinejs/alpine/discussions/1594

**Comment by Luddinus** [comment]
@peterhijma try this: https://github.com/alpinejs/alpine/discussions/1594#discussioncomment-906534
